x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -770,7 +770,17 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                     .build(),
                 // For ILM policy for APM & Endpoint packages that have delete action
                 RoleDescriptor.IndicesPrivileges.builder()
-                    .indices(".logs-endpoint.diagnostic.collection-*", "traces-apm.sampled-*")
+                    .indices(
+                        ".logs-endpoint.diagnostic.collection-*",
+                        "logs-apm.app-*",
+                        "logs-apm.error-*",
+                        "metrics-apm.app.*",
+                        "metrics-apm.internal-*",
+                        "metrics-apm.profiling-*",
+                        "traces-apm-*",
+                        "traces-apm.rum-*",
+                        "traces-apm.sampled-*"
```

Comment:
Do we need to be that specific? If we went ahead with `logs-apm.*`,  `logs-apm-*`, `traces-apm.*`, `traces-apm-*` ,  `metrics-apm.*` and `metrics-apm-*` we'd lower the risk of missing data streams in the future and be more aligned with [current definitions](https://github.com/elastic/elasticsearch/blob/ce5c63508a4b0da52bd4a6e8dd289bb2bf656431/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java#L257-L274).

Reply:
Good call. I'll update to that.